### ISSUE TITLE
WHCM styles improvements

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail.org#454]
Based on the audit from @shyusu4, see [Wagtail.org | Accessibility audit spreadsheet](https://docs.google.com/spreadsheets/d/1kWb6f9YlT-UA2TC4ioAH40Y5T18hd_Hbt_ncTamabW4/edit). WHCM fixes:

## Issues

- [x] Close icon in the search modal isn’t visible.
- [x] Menu toggle isn't visible.
- [x] Copy icon in "pip install Wagtail" button disappears when clicking on it.
- [x] (not in audit) SVG icons in buttons now match the button text color
- [x] Social media icons are not using the correct color (linktext)
- [x] All social media, “sponsor wagtail”, and “contribute code” icons aren't visible.
- [ ] #463
- [x] Arrow icon within CTA button isn't visible when hovered.
- [x] Arrow icon within CTA link in sub-navigation panel isn’t visible.
- [x] (Blog posts) Code blocks should have a clear separation from regular content.

## Working on this

Assigning this to @shyusu4 as part of her Outreachy project

**Comment by thibaudcolas** [comment]
Closing with last remaining item at #463.
